test/unit_tests/unit_test_math.cpp

```diff
@@ -114,6 +114,50 @@ namespace pvmath
         }
         return true;
     }
+
+    template <typename _Type>
+    bool pointProjection()
+    {
+        for ( uint32_t i = 0; i < Unit_Test::runCount(); ++i ) {
+            const PointBase2D<_Type> testPoint( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );
+
+            const PointBase2D<_Type> pointBase( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );
+            const PointBase2D<_Type> pointBaseX( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), pointBase.y );
+            const PointBase2D<_Type> pointBaseY( pointBase.x, Unit_Test::randomFloatValue<_Type>( -330, 330, 0.01f ) );
+
+            const LineBase2D<_Type> lineX( pointBase, pointBaseX );
+            const LineBase2D<_Type> lineY( pointBase, pointBaseY );
+
+            const PointBase2D<_Type> resultPointX( testPoint.x, pointBase.y );
+            const PointBase2D<_Type> resultPointY( pointBase.x, testPoint.y );
+
+            if ( !( lineX.projection( testPoint ) == resultPointX ) || !( lineY.projection( testPoint ) == resultPointY ) )
```

Comment:
`if ( ( lineX.projection( testPoint ) != resultPointX ) || ( lineY.projection( testPoint ) != resultPointY ) )` is simpler :)

Reply:
Of course.
I just was a bit lazy to implement `bool operator != ( const PointBase2D & point ) const `.